main.go

```diff
@@ -454,8 +467,14 @@ func serverListen(context *Context) error {
 		return err
 	}
 
+	serverConfig, ok := context.tlsConfigSource.GetServerConfig(config)
+	if !ok {
+		logger.Print("TLS config not appropriate for server role")
+		return errors.New("TLS config not appropriate for server role")
```

Comment:
Should it be possible for this to ever happen? What would cause this state?

Reply:
This is defensive. Technically as written it isn't possible since even though `buildCertificate` can return `NoCertificate` the call to `serverValidateFlags` after the fact ensures that the right flags were provided before `serverListen` is called.